docs/markdown/Wrap-dependency-system-manual.md

```diff
@@ -135,7 +135,7 @@ case, the directory will be copied into `subprojects/` before applying patches.
 - `push-url` - alternative url to configure as a git push-url. Useful if
   the subproject will be developed and changes pushed upstream.
   *(since 0.37.0)*
-- `clone-recursive` - also clone submodules of the repository
+- `clone-recursive = true` - also clone submodules of the repository
```

Comment:
This is interesting to think about because elsewhere we don't describe the expected type to use which makes this stand out. Also, it feels awkward given that false could be used instead?

Reply:
I think people will assume that omitting the option completely assumes false.

Reminds me of how I designed [canfigger](https://github.com/andy5995/canfigger/) though. Options are basically togged "on" by adding the option names with no value. Since, like in this case, being omitted assumes "false".

Reply:
Option names with no value are just an error though?

Reply:
> Option names with no value are just an error though?

No. The developer using canfigger can check if the option/key exist and do whatever based on that. 

If the developer doesn't check for a key that's in a config file, it's just ignored. Or in the case of 'rmw' (which is also one of my projects that uses canfigger), I issue a warning about an invalid option.

Reply:
Option names (in meson) with no value are just an error though?

Reply:
Yes.

```
meson.build:1:0: ERROR: Failed to parse /home/andy/src/rmw-project/rmw/subprojects/decimal.wrap: Source contains parsing errors: '/home/andy/src/rmw-project/rmw/subprojects/decimal.wrap'
	[line  7]: 'clone-recursive\n'
```

Reply:
I feel like you're probably missing the point I was trying to make -- I wasn't asking a question, and my attempt at r repeating my original comment with a clarification in parentheses was supposed to make it more obvious that I wasn't asking a question. :)

Reply:
Ok. I thought you were repeating your question with the clarification in parentheses because I first interpreted your question as asking about canfigger. And then you asked again and put "in meson" in parentheses. I thought you were clarifying that your question was about a meson.

I was a bit confused that you already didn't know the answer. :) So... what was your point? You were trying to tell me that Option names with no value are just an error and there's no plans right now to change that?

Reply:
There are indeed no plans to change that. In particular, meson uses python's configparser format (and it is a hard requirement that it cannot use dependencies which aren't present in the python standard library).

configparser will simply raise a fatal error if you specify `clone-recursive` without a value.


```
meson.build:1:0: ERROR: Failed to parse /tmp/fakeproj/subprojects/foo.wrap: Source contains parsing errors: '/tmp/fakeproj/subprojects/foo.wrap'
	[line  3]: 'clone-recursive\n'
```